polus-csv-statistics-plugin/src/main.py

```diff
@@ -0,0 +1,217 @@
+        return
+    std(data_list,data_dict)
+    try:
+        data_dict['skew'] = sum([(x-data_dict['mean'])**3 for x in data_list])/(data_dict['count']*data_dict['std']**3)
+    except ZeroDivisionError:
+        data_dict['skew'] = 'NaN'
+    return data_dict
+    
+def kurtosis(data_list,data_dict):
+    if 'kurt' in data_dict.keys():
+        return
+    std(data_list,data_dict)
+    try:
+        data_dict['kurt'] = sum([(x-data_dict['mean'])**4 for x in data_list])/(data_dict['count']*data_dict['std']**4) - 3
+    except ZeroDivisionError:
+        data_dict['kurt'] = 'NaN'
+    return data_dict
+    
+# Dictionary of input statistics
+STATS = {'mean': mean,
+         'median': median,
+         'std': std,
+         'var': var,
+         'skew': skewness,
+         'kurt': kurtosis,
+         'count': count,
+         'max': maxval,
+         'min': minval}
+
+if __name__=="__main__":
+    # Initialize the logger
+    logging.basicConfig(format='%(asctime)s - %(name)-8s - %(levelname)-8s - %(message)s',
+                        datefmt='%d-%b-%y %H:%M:%S')
+    logger = logging.getLogger("main")
+    logger.setLevel(logging.INFO)
+
+    # Setup the argument parsing
+    logger.info("Parsing arguments...")
+    parser = argparse.ArgumentParser(prog='main', description='Calculate simple statistics to groups of data in a csv file.')
+    parser.add_argument('--statistics', dest='statistics', type=str,
+                        help='Types of statistics to calculate', required=True)
+    parser.add_argument('--inpDir', dest='inpDir', type=str,
+                        help='Input csv collection to be processed by this plugin', required=True)
+    parser.add_argument('--outDir', dest='outDir', type=str,
+                        help='Output collection', required=True)
+    
+    # Parse the arguments
+    args = parser.parse_args()
+    statistics = args.statistics.split(',')
+    logger.info('statistics = {}'.format(statistics))
+    inpDir = args.inpDir
+    logger.info('inpDir = {}'.format(inpDir))
+    outDir = args.outDir
+    logger.info('outDir = {}'.format(outDir))
+    
+    # Get a list of all input files
+    csv_files = [f for f in Path(inpDir).iterdir() if f.name.endswith('csv')]
+    
+    # Open each csv files
+    for feat_file in csv_files:
+        fpath = str(feat_file.absolute())
+        out = str(Path(outDir).joinpath(feat_file.name).absolute())
+        with open(fpath,'r') as fr:
+            with open(str(out),'w') as fw:
+                # Read the first line, which should contain headers
+                first_line = fr.readline()
+                headers = first_line.rstrip('\n').split(',')
+                var_ind = {key:val for key,val in enumerate(headers)} # map headers to line positions
+                
+                # If no column is labeled file, throw an error
+                if 'file' not in headers:
+                    ValueError('At least one column must have a header title file.')
```

Comment:
should this be an error or a warning? The assumption would be with no header you could still do column wise math with no header but it wouldn't be ideal.